### PR TITLE
[autoscaler] fix  too many values to unpack (expected 2) bug

### PR DESCRIPTION
**Pull request by scv119**
autoscaler_summary.pending_launches is a dict thus causing the iteration fails with `too many values to unpack` issue.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]

File: python/ray/autoscaler/_private/monitor.py [hunk 1]
```diff
@@ -400,11 +400,15 @@ def _run(self):
...
-                        self.autoscaler.all_node_types,
-                    )
+                    try:
+                        self.emit_metrics(
+                            load_metrics_summary,
+                            autoscaler_summary,
+                            self.autoscaler.all_node_types,
+                        )
+                    except Exception:
+                        logger.error("Error emitting metrics")
```
Comment (aslonnie) [inline python/ray/autoscaler/_private/monitor.py:410]: use `logger.exception` ? or print the stack trace?

Comment (aslonnie) [reply python/ray/autoscaler/_private/monitor.py:410]: I don't think we want to drop the exception silently.

**Review by aslonnie — APPROVED** [review]

**Comment by aslonnie** [comment]
@scv119 , could you also get a cherrypick PR for branch `releases/2.5.0`

**Comment by ArturNiederfahrenhorst** [comment]
The 2.5.1 branch is up, please also create a cherry pick PR for that one.

**Pull request merged**